src/main/kotlin/org/jitsi/nlj/RtpLayerDesc.kt

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.nlj
+
+import org.jitsi.nlj.rtp.VideoRtpPacket
+import org.jitsi.nlj.rtp.codec.vp8.Vp8Packet
+import org.jitsi.nlj.stats.NodeStatsBlock
+import org.jitsi.utils.stats.RateStatistics
+
+/**
+ * Keeps track of its subjective quality index,
+ * its last stable bitrate and other useful things for adaptivity/routing.
+ *
+ * @author George Politis
+ */
+class RtpLayerDesc
+constructor(
+    /**
+     * The index of this instance's encoding in the source encoding array.
+     */
+    val eid: Int,
+    /**
+     * The temporal layer ID of this instance.
+     */
+    val tid: Int,
+    /**
+     * The spatial layer ID of this instance.
+     */
+    val sid: Int,
+    /**
+     * The max height of the bitstream that this instance represents. The actual
+     * height may be less due to bad network or system load.
+     */
+    // XXX we should be able to sniff the actual height from the RTP
```

Comment:
Does this comment still apply?

Reply:
For now, yes.  I want to separate the change to set that value from this refactor.